Modules/Test/classes/CanAccessFileUploadAnswer.php

```diff
@@ -106,35 +106,40 @@ private function accessCodeOk(string $file, int $test_id) : bool
     private function userDidUpload(int $test_id, string $file, string $code = null) : bool
     {
         $where = [
+            'active_id = active_fi',
             'user_fi = %s',
             'value1 = %s',
             'anonymous_id ' . (null === $code ? 'IS' : '=') . ' %s',
             'test_fi = %s',
         ];
 
         $result = $this->container->database()->queryF(
-            'SELECT 1 FROM tst_solutions INNER JOIN tst_active ON active_id = active_fi WHERE ' . implode(' AND ', $where),
+            'SELECT 1 FROM tst_solutions WHERE EXISTS (SELECT 1 FROM tst_active WHERE ' . implode(' AND ', $where) . ')',
             ['integer', 'text', 'text', 'integer'],
             [$this->container->user()->getId(), $file, $code, $test_id]
         );
 
         return (bool) $this->container->database()->numRows($result);
     }
 
-    private function activeIdOfFile(string $file) : ?int
+    private function activeIdOfFile(string $file, int $test) : ?int
     {
+        $is_upload_question = 'EXISTS (SELECT 1 FROM qpl_qst_type INNER JOIN qpl_questions ON question_id = question_fi WHERE type_tag = "assFileUpload" AND question_type_id = question_type_fi)';
```

Comment:
Please use `$this->container->database()->quote('assFileUpload', 'text');` here (and in the other PR).

Reply:
Other PR is adjusted as well.